subprojects/model-core/src/main/java/org/gradle/internal/instantiation/generator/ManagedObjectFactory.java

```diff
@@ -111,22 +113,31 @@ public Object newInstance(ModelObject owner, String propertyName, Class<?> type,
     private static ManagedPropertyName displayNameFor(ModelObject owner, String propertyName) {
         if (owner.getModelIdentityDisplayName() instanceof ManagedPropertyName) {
             ManagedPropertyName root = (ManagedPropertyName) owner.getModelIdentityDisplayName();
-            return new ManagedPropertyName(root.owner, root.propertyName + "." + propertyName);
+            return new ManagedPropertyName(root.ownerDisplayName, root.propertyName + "." + propertyName);
```

Comment:
This changes the behavior by making the display name String computation eager.

How about making the display name a `Cached` value? See `org.gradle.internal.serialization.Cached#of`.

Reply:
Indeed it changed. Thanks for the suggestion, I did it, PTAL

Reply:
Looks good 👍